### DISCUSSION TITLE
Pmetrics installation

### DISCUSSION THREAD

**Discussion by BoburaII** [LAPKB/Pmetrics#240] [General]
Hello, 
I've updated MacOS and run into trouble with Pmetrics. Attempts to re-install have been unsuccessful, with the following error message: 

* installing *source* package ‘Pmetrics’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘ITex’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘ITex’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘NPex’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘NPex’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘NPex_val’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘NPex_val’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘badData’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘badData’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘dataEx’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘dataEx’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘modEx’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘modEx’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘modelLibrary’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘modelLibrary’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘simEx’
Warning: namespace ‘Pmetrics’ is not available and has been replaced
by .GlobalEnv when processing object ‘simEx’
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘Pmetrics’ in library.dynam(lib, package, package.lib):
 shared object ‘Pmetrics.so’ not found
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/Pmetrics’
Warning message:
In i.p(...) :
  installation of package ‘/var/folders/5z/2ms25s0d3nj5thkpd0bvlqk40000gn/T//RtmpcSB12m/file12754f1ea749/Pmetrics_2.1.1.tar.gz’ had non-zero exit status

I seem not to figure out the error. Would appreciate help with a work around.
Thanks

**Comment by ChrisKonig** [comment]
I have the same issue during installation on my Mac (Sanoma 14.1.2). Would appreciate help to get it to work. 

Thanks, 
Chris

**Comment by mnneely** [comment]
Sorry for the trouble. We’ll look into it and come up with a solution. I have an idea where to look. We’re trying to get the package “CRAN-worthy” so it can be distributed in the normal fashion. This is a HUGE undertaking as CRAN is very strict. One of the issues is the inclusion of all the sample data, which make the package exceed the usual limitations on CRAN. I’m trying different solutions, including separating the data into a separate package. This error may have crept in because of that.

**Reply by ChrisKonig** [reply]
Is the old manual way of installation (with importing the package .tgz) still possible? I guess not optimal, but could be a quick fix for me. Thank you for looking and helping us with this.

**Comment by mnneely** [comment]
I created an [issue](https://github.com/LAPKB/Pmetrics/issues/242#issue-2662876018) to track this.  I also believe I've fixed the issue, which was not related to the consideration I posted above.  I'm going to close this discussion. Any other points, please add them to the issue thread. Thanks!
